skrub/_datetime_encoder.py

```diff
@@ -323,3 +324,11 @@ def _check_params(self):
             raise ValueError(
                 f"'resolution' options are {allowed}, got {self.resolution!r}."
             )
+
+    def _more_tags(self):
+        return {"preserves_dtype": []}
+
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        tags.transformer_tags = TransformerTags(preserves_dtype=[])
```

Comment:
does this class not inherit from `TransformerMixin`? Why not 

```suggestion
        tags.transformer_tags.preserves_dtype = []
```

to not override all the other tags from the parent classes?

Reply:
We get into trouble with the `set_output` for this one: https://github.com/skrub-data/skrub/pull/1135#discussion_r1837807800

Reply:
Actually this is not mentioned in the comment. @jeromedockes can probably offer the detailed perspective.

Reply:
I understand that comment as "we don't need it, we don't have to use it". Now with tags, you kinda better use them, and if there's something you don't like from that mixin, you can always override.

Reply:
> Now with tags, you kinda better use them, and if there's something you don't like from that mixin, you can always override.

I recall that the discussion was around the `set_output` (we need @jeromedockes for the details but it might be related to avoiding any data copy or dtype conversion - keep the feature of the container library in short). We are fine having the tags, but inheriting from the `TransformerMixin` brings the all the infrastructure of `set_output`. Overwriting the inheritance is not easy while redefining the `Tags` that as in `TransformerMixin` is easy.

Reply:
If wrapping with `set_output` is adding copies, we should fix that.

Reply:
it does not add copies, at least when the output is already in the type of container the configuration is asking for. (AFAICT)